### PR TITLE
[Tools] GstPipe<-->Pbtxt Conversion / Skeleton

### PR DESCRIPTION
**Pull request by myungjoo**
Skeleton code for GstPipe<-->Pbtxt Conversion.

Ultimately, we can use web-based WYSIWYG tools based on this!

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: tools/development/gstPrototxt.py
Comment (dongju-chae) [inline tools/development/gstPrototxt.py:90]: Where does `a` come from?

Comment (myungjoo) [reply tools/development/gstPrototxt.py:90]: Yes.. it should've been f.

File: tools/development/gstPrototxt.py
Comment (kparichay) [inline tools/development/gstPrototxt.py:2]: for scripts, SPDX can be at max at line 2 (https://lore.kernel.org/patchwork/patch/883424/)

File: tools/development/gstPrototxt.py
Comment (kparichay) [inline tools/development/gstPrototxt.py:23]: this is a global constant than global variable right? How about CAPITALIZING it?

File: tools/development/gstPrototxt.py
Comment (kparichay) [inline tools/development/gstPrototxt.py:47]: this increases the name counter for differently named variables. Is this intended?

File: tools/development/gstPrototxt.py
Comment (kparichay) [inline tools/development/gstPrototxt.py:49]: bring this outside the `if else` and add this name to `declaredNames`?

File: tools/development/gstPrototxt.py
Comment (kparichay) [inline tools/development/gstPrototxt.py:74]: in python , its recommended to use `filter` than use a loop.
also, as nameList stores all the names of filters, checking directly in nameList would also suffice.

File: tools/development/gstPrototxt.py
Comment (myungjoo) [reply tools/development/gstPrototxt.py:23]: Applied.

File: tools/development/gstPrototxt.py
Comment (myungjoo) [reply tools/development/gstPrototxt.py:2]: Applied.

File: tools/development/gstPrototxt.py
Comment (myungjoo) [reply tools/development/gstPrototxt.py:49]: No, the intention is to manage filters with its id declared (```filtername id=THENAME```) and filters without such (generated) differently. If it is generated (not declared), we can hide the id in GUI as well. If it is declared, we need to show the id in GUI.

File: tools/development/gstPrototxt.py
Comment (myungjoo) [reply tools/development/gstPrototxt.py:74]: Updated.

**Review by dongju-chae — APPROVED** [review]
LGTM, after this PR is merged, I will take over this topic.

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

**Pull request merged**
